docker/nginx/conf.d/client.conf

```diff
@@ -151,12 +151,36 @@ server {
 			-- example response: '{"skylink":"sia://XABvi7JtJbQSMAcDwnUnmp2FKDPjg8_tTTFP4BwMSxVdEg"}'
 			local hnsres_json = json.decode(hnsres_res.body)
 
-			-- try to match the skylink with sia:// prefix
-			local skylink, skylink_rest = string.match(hnsres_json.skylink, "sia://([^/?]+)(.*)")
+			if hnsres_json.skylink then
+				-- try to match the skylink with sia:// prefix
+				local skylink, skylink_rest = string.match(hnsres_json.skylink, "sia://([^/?]+)(.*)")
+
+				-- in case the skylink did not match, assume that there is no sia:// prefix and try to match again
+				if skylink == nil then
+					skylink, skylink_rest = string.match(hnsres_json.skylink, "/?([^/?]+)(.*)")
+				end
+			end
+
+			if hnsres_json.registry then
+				local publickey = hnsres_json.registry.publickey
+				local datakey = hnsres_json.registry.datakey
+
+				-- make a get request to /skynet/registry endpoint with the credentials from text record
+				local registry_res = ngx.location.capture("/skynet/registry?publickey=" .. publickey .. "&datakey=" .. datakey)
```

Comment:
Now I get it why you bolded the "should be uri-encoded". This is a little uncomfortable but I don't see any major damage that can happen from it aside from a bad request.

Reply:
Actually I think we should revisit the portal code at some point to ensure all the user input is properly escaped, sanitised  or regex matched. It's kind of a MVP implementation while we are still experimenting with the features.

Reply:
Can you open a ticket for that? Otherwise it would surely be forgotten.

Reply:
Also, meaningful error messages would help, we now return 404 if anything is not set up correctly. That's a todo on my list at some point.